### PR TITLE
Add doc example to HashMap::hasher

### PR DESCRIPTION
**Pull request by napen123**

### CODE REVIEW

File: src/libstd/collections/hash/map.rs [hunk 1]
```diff
@@ -691,6 +691,16 @@ impl<K, V, S> HashMap<K, V, S>
     /// Returns a reference to the map's [`BuildHasher`].
     ///
     /// [`BuildHasher`]: ../../std/hash/trait.BuildHasher.html
+    ///
+    /// # Examples
```
Comment (frewsxcv) [inline src/libstd/collections/hash/map.rs]: I think you might need a newline after this for rustdoc+markdown to treat it as a doc test

**Review by frewsxcv — APPROVED** [review]
looks good! just one small comment about whitespace

**Comment by GuillaumeGomez** [comment]
Now please [squash](https://blog.guillaume-gomez.fr/Git-tips/1/2) your commits and we're good to go!

**Comment by frewsxcv** [comment]
thanks for your contribution @napen123! 🎉 

@bors r+ rollup

**Comment by Mark-Simulacrum** [comment]
Merged in https://github.com/rust-lang/rust/pull/44936.

**Pull request closed without merging**